services/management/backup/artifacts_service.go

```diff
@@ -96,6 +100,90 @@ func (s *ArtifactsService) ListArtifacts(context.Context, *backupv1beta1.ListArt
 	}, nil
 }
 
+// DeleteArtifact deletes specified artifact.
+func (s *ArtifactsService) DeleteArtifact(
+	ctx context.Context,
+	req *backupv1beta1.DeleteArtifactRequest,
+) (*backupv1beta1.DeleteArtifactResponse, error) {
+	var s3Config *models.S3LocationConfig
+	var artifactName string
+	if err := s.db.InTransaction(func(tx *reform.TX) error {
```

Comment:
why are there 2 transactions? each one is finding artifact. Can't we merge it?

Reply:
fixed